roscala/src/main/scala/coop/rchain/rosette/prim/rblstring.scala

```diff
@@ -207,6 +207,53 @@ object rblstring {
     }
   }
 
+  /**
+    * Define the string-get-token primitive.
+    * This returns the nth substring delimited by one of the separators
+    * in the second string.
+    *
+    * examples:
+    * (string-get-token "aX,bY.cZ-eW=fQ/gC" 2 "/-=,.") ====> "cZ"
+    */
+  object stringGetToken extends Prim {
+    override val name: String = "string-get-token"
+    override val minArgs: Int = 3
+    override val maxArgs: Int = 3
+
+    @checkArgumentMismatch
+    override def fn(ctxt: Ctxt): Either[PrimError, RblString] = {
+      val elem = ctxt.argvec.elem
+
+      for {
+        str <- checkType[RblString](0, elem) // Ensure arg0 is a RblString
+        w   <- checkType[Fixnum](1, elem)    // Ensure arg1 is a Fixnum
+        sep <- checkType[RblString](2, elem) // Ensure arg2 is a RblString
+      } yield {
+        // The C++ Rosette exhibits some weird edge conditions.
```

Comment:
Can you give us some examples?

Reply:
I started to get these and realized I didn't understand some aspects of how it operates.  I'm going to cancel the PR and work on it a while longer.  Sorry.